objects/Master.py

```diff
@@ -0,0 +1,66 @@
+import sys
+import shutil
+import os
+import asyncio
+from hyperopt import fmin, tpe
+from hyperopt.mongoexp import MongoTrials
+from .OPTSubprocess import OPTSubprocess
+# import from directory up
+sys.path.append('..')
+from NWT_SUBMIT.NWTOPT_FILES.objective import objective
+from config.HParams import MF5HParams
+
+class Master(OPTSubprocess):
+    def __init__(self, id, logger, cwd, ip, port, key, random, trials):
+        super().__init__('master', id, logger)
+        self.cwd = cwd
+        try:
+            os.mkdir(os.path.join(self.cwd, 'NWT_SUBMIT', 'PROJECT_FILES', 'nwts'))
+        except FileExistsError:
+            shutil.rmtree(os.path.join(self.cwd, 'NWT_SUBMIT', 'PROJECT_FILES', 'nwts'))
+            os.mkdir(os.path.join(self.cwd, 'NWT_SUBMIT', 'PROJECT_FILES', 'nwts'))
+
+        self.hparams = MF5HParams().hparams
+        self.ip = ip
+        self.port = port
+        self.key = key
+        self.random = random
+        self.mongoTrials = None
+        self.trials = trials
+
+        try:
+            os.remove(os.path.join(self.cwd, 'NWT_SUBMIT', 'PROJECT_FILES', 'nwts', 'nwtnum.txt'))
```

Comment:
again, wouldn't rely on try/except here. You can just test to see if the file is there and, if it is, delete it

Reply:
I think you are looking at old commit still, new one for me has path.exists all over.